### PR TITLE
Update to new azure-sdk-for-js

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
Fixes https://github.com/microsoft/vscode-azureappservice/issues/904

### CODE REVIEW

File: src/commands/createWebApp/setPostPromptDefaults.ts [hunk 1]
```diff
@@ -78,12 +78,7 @@ export async function setPostPromptDefaults(wizardContext: IAppServiceWizardCont
     }
 }
 
-export async function getAppServicePlan(wizardContext: IAppServiceWizardContext, rgName: string, newPlanName: string): Promise<AppServicePlan | null> {
-    const client: WebSiteManagementClient = createAzureClient(wizardContext, WebSiteManagementClient);
-    return await client.appServicePlans.get(rgName, newPlanName);
-}
-
-function checkPlanForPerformanceDrop(asp: AppServicePlan): boolean {
+function checkPlanForPerformanceDrop(asp: WebSiteManagementModels.AppServicePlan): boolean {
```
Comment (nturinski) [inline src/commands/createWebApp/setPostPromptDefaults.ts:81]: Just to make sure, does` client.appServicePlans.list` now also return the asp's `numberOfSites` property?

File: src/commands/postDeploy/validateWebSite.ts [hunk 2]
```diff
@@ -54,14 +47,10 @@ export async function validateWebSite(deploymentCorrelationId: string, siteTreeI
...
-                request.resolveWithFullResponse = true;
-                const response = await requestUtils.sendRequest<IncomingMessage>(request);
-                currentStatusCode = response.statusCode;
+                const response: HttpOperationResponse = await client.sendRequest({ method: 'GET', url });
+                currentStatusCode = response.status;
             } catch (error) {
-                // tslint:disable-next-line:strict-boolean-expressions
-                const response = (<WebError>error).response || {};
-                currentStatusCode = isNumber(response.statusCode) ? response.statusCode : 0;
+                currentStatusCode = error instanceof RestError ? error.statusCode : 0;
```
Comment (nturinski) [inline src/commands/postDeploy/validateWebSite.ts:53]: The behavior seems a little different here.  Did we only want to record RestError status codes?

File: src/explorer/trialApp/TrialAppClient.ts [hunk 3]
```diff
@@ -65,34 +63,35 @@ export class TrialAppClient implements ISimplifiedSiteClient {
...
+        });
     }
 
-    public async listApplicationSettings(): Promise<StringDictionary> {
+    public async listApplicationSettings(): Promise<WebSiteManagementModels.StringDictionary> {
         const kuduClient: KuduClient = await this.getKuduClient();
-        const settings: StringDictionary = {};
-        settings.properties = <{ [name: string]: string }>await kuduClient.settings.getAll();
+        const settings: WebSiteManagementModels.StringDictionary = {};
+        settings.properties = <{ [name: string]: string }>(await kuduClient.settings.getAll()).body;
```
Comment (nturinski) [inline src/explorer/trialApp/TrialAppClient.ts:89]: Why did the response type here change?

**Review by nturinski — APPROVED** [review]

File: src/commands/postDeploy/validateWebSite.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply src/commands/postDeploy/validateWebSite.ts:53]: I think the difference you're noticing just comes down to "request-promise" vs. "@azure/ms-rest-js". If there is a statusCode in the error, it'll be a `RestError` - that's just how "@azure/ms-rest-js" works. I validated a few sites with successful status codes (2xx) and a few with failed status codes (4xx) and we were working the same

File: src/commands/createWebApp/setPostPromptDefaults.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/commands/createWebApp/setPostPromptDefaults.ts:81]: Discussed offline

**Pull request merged**